staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -1481,7 +1481,6 @@ func RunWatchSemantics(ctx context.Context, t *testing.T, store storage.Interfac
 			// make sure we only get initial events
 			testCheckResultsInStrictOrder(t, w, scenario.expectedInitialEvents(createdPods))
 			testCheckResultsInStrictOrder(t, w, scenario.expectedInitialEventsBookmark(createdPods))
-			testCheckNoMoreResults(t, w)
```

Comment:
It seems there is no need to ensure there won't be additional data because `testCheckResultsInStrictOrder` (line 1494) will fail if there are any, and in case `expectedEventsAfterEstablishingWatch` is empty, the final call to `testCheckNoMoreResults` (line 1495) will fail.

Reply:
with this change the running time is 

```
ok  	k8s.io/apiserver/pkg/storage/cacher	25.079s
```

previously it was:
```
ok  	k8s.io/apiserver/pkg/storage/cacher	43.588s
```

Reply:
> It seems there is no need to ensure there won't be additional data because testCheckResultsInStrictOrder (line 1494) will fail if there are any

this isn't true - if the events returned in watch are exactly as expected and some garbage at the end, testCheckResultsInStrictOrder will pass

Reply:
since we don't close the watch then the garbage at the end will be picked by the next call to `testCheckResultsInStrictOrder` (line 1494) and the test will fail.

Reply:
ahh - ok, that makes sense

and you're not touching the `testCheckNoMoreResults` at the end of the test.

So yes - you're right, it's fine.